### PR TITLE
Enable SourceLink and update package metadata

### PR DESCRIPTION
**Pull request by seif**
Update nuget metadata to follow best practices set out in https://docs.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices
Enable symbol source and deterministic builds https://mitchelsellers.com/blog/article/net-5-deterministic-builds-source-linking
Bump to version 1.0.0-RC and update to use VersionPrefix and Suffix and let the framnework generate assembly version, file version and package versions.

### CODE REVIEW

**Review by sgerrand — APPROVED** [review]
LGTM, no blockers. 👍

File: Duffel.ApiClient/Duffel.ApiClient.csproj [hunk 1]
```diff
@@ -3,21 +3,35 @@
...
+        <Copyright>Duffel Technology Ltd. 2022</Copyright>
         <Company>Duffel Technology Ltd.</Company>
-        <PackageVersion>0.4.0</PackageVersion>
-        <Title>dotNet client for interacting with the Duffel API</Title>
+        <VersionPrefix>1.0.0</VersionPrefix>
+        <VersionSuffix>RC</VersionSuffix>
+        <Title>Duffel.ApiClient</Title>
+        <Description>.NET client for interacting with the Duffel API https://docs.duffel.com/</Description>
+        <PackageId>Duffel.ApiClient</PackageId>
+        <PackageIcon>logo_128.png</PackageIcon>
```
[the remark this answers is not in the document]
Comment (seif) [reply Duffel.ApiClient/Duffel.ApiClient.csproj:14]: Image is small, dont think we need the Git LFS

File: Duffel.ApiClient/Duffel.ApiClient.csproj [hunk 2]
```diff
@@ -3,21 +3,35 @@
...
         <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-        <Authors>Duffel Technology Ltd.</Authors>
-        <Copyright>Duffel Technology Ltd.</Copyright>
+        <Authors>Duffel</Authors>
+        <Copyright>Duffel Technology Ltd. 2022</Copyright>
         <Company>Duffel Technology Ltd.</Company>
-        <PackageVersion>0.4.0</PackageVersion>
-        <Title>dotNet client for interacting with the Duffel API</Title>
+        <VersionPrefix>1.0.0</VersionPrefix>
+        <VersionSuffix>RC</VersionSuffix>
```
[the remark this answers is not in the document]
Comment (seif) [reply Duffel.ApiClient/Duffel.ApiClient.csproj:10]: What are you saying @sgerrand :D

File: Duffel.ApiClient/Duffel.ApiClient.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgerrand) [reply Duffel.ApiClient/Duffel.ApiClient.csproj:14]: Fair enough!

**Review by sgerrand — APPROVED** [review]
Still LGTM. 👍

**Pull request merged**